### PR TITLE
Phantom

### PR DESCRIPTION
**Pull request by RaphaelBonatti**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/database/firebase/api/CommonDatabaseAPI.java [hunk 1]
```diff
@@ -70,6 +70,8 @@
      */
     void sendUserPosition(PlayerForFirebase playerForFirebase);
 
+    void sendUserPhantom(PlayerForFirebase playerForFirebase);
```
Comment (kangpl) [inline app/src/main/java/ch/epfl/sdp/database/firebase/api/CommonDatabaseAPI.java:73]: JavaDoc needed

File: app/src/main/java/ch/epfl/sdp/database/firebase/api/Function.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package ch.epfl.sdp.database.firebase.api;
+
+public interface Function<T, R>  {
```
Comment (kangpl) [inline app/src/main/java/ch/epfl/sdp/database/firebase/api/Function.java:3]: JavaDoc needed

File: app/src/main/java/ch/epfl/sdp/game/Server.java [hunk 3]
```diff
@@ -77,7 +76,8 @@ public void update() {
             sendGameArea();
             sendEnemies();
             sendItemBoxes();
-            sendPlayersHealth();
+            //sendPlayersHealth();
```
Comment (kangpl) [inline app/src/main/java/ch/epfl/sdp/game/Server.java:79]: This line can be removed

File: app/src/main/java/ch/epfl/sdp/item/ItemFactory.java [hunk 4]
```diff
@@ -11,8 +11,8 @@ public Item getItem(String itemName) {
                 return new Shield(Integer.parseInt(parts[1]));
             case "Shrinker":
                 return new Shrinker(Integer.parseInt(parts[1]), Integer.parseInt(parts[2]));
-            case "Scan":
-                return new Scan(Integer.parseInt(parts[1]));
+            case "Phantom":
+                return new Phantom(Integer.parseInt(parts[1]));
```
Comment (kangpl) [inline app/src/main/java/ch/epfl/sdp/item/ItemFactory.java:15]: Good to replace Scan with Phantom，which adds more fun to multi-player mode

File: app/src/main/java/ch/epfl/sdp/artificial_intelligence/Movement.java [hunk 5]
```diff
@@ -38,24 +36,6 @@ public void setVelocity(float velocity) {
         this.velocity = velocity/GameThread.FPS;
     }
 
-    /**
-     * Gets the current acceleration determined by the movement.
-     *
-     * @return A value representing the acceleration of the movement.
-     */
-    public float getAcceleration() {
```
Comment (Zeavee) [inline app/src/main/java/ch/epfl/sdp/artificial_intelligence/Movement.java:46]: Thank you for removing useless parameters

File: app/src/main/java/ch/epfl/sdp/database/firebase/api/ClientDatabaseAPI.java [hunk 6]
```diff
@@ -62,5 +62,9 @@
      */
     void cleanListeners();
 
+    /**
+     * This method add a listener to the signal that indicates whether the server is alive or not.
+     * @param onValueReadyCallback Callback when the signal changed in the Firebase FireStore
+     */
```
Comment (Zeavee) [inline app/src/main/java/ch/epfl/sdp/database/firebase/api/ClientDatabaseAPI.java:68]: Javadoc!

**Pull request merged**